### ISSUE TITLE
require('stre'+'am') doesn't do what I thought it did

### ISSUE THREAD

**Issue by calvinmetcalf** [nodejs/readable-stream#260]
instead of only including stream if somebody else included it it apparently just doesn't include it in browserify vs node.  This tells us a couple things

1. nobody is doing readable-stream instance of require('stream') in the browser
2. we could theoretically just replace this with a "browser":{"stream":false}
3. webpack is doing what I thought browserify did, but they are doing it wrong

**Comment by calvinmetcalf** [comment]
see webpack/webpack#4378

**Comment by mcollina** [comment]
Thanks @calvinmetcalf for taking this heads-on.
I'm 👍  on `"browser": { "stream": false }`.

Should we discuss this on the next wg meeting and release this ASAP?

**Comment by calvinmetcalf** [comment]
yeah

**Comment by parshap** [comment]
I've struggled with the `require('st'+'ream')` calls while trying to get readable-stream to run in React Native. Here's what I've observed:

1. **When using Browserify, `require('st'+'ream')` throws an exception.** This is because Browserify *does not* resolve `'st' + 'ream'` at compile-time, so the module is not available at run-time. This is true even when another module has required `stream` before because the `require` function in Browserify is local to the module and only gives access to the dependencies resolved at compile-time.

1. **When using React Native Packager, `require('st'+'ream')` returns an empty object `{}`.** This is because React Native Packager *does* resolve `'st' + 'ream'` at compile-time and [`stream: false` is set in `package.json`][stream-false], which [according to the `package.json` `browser` field spec][browser spec], should cause the module to resolve to an empty object `{}`.

This results in readable-stream throwing an exception and crashing the app. My current solution is to [patch readable-stream to remove `require('st' + 'ream')` calls][patch] in a `postinstall` hook.

Going forward, my suggestion is to:

1. Break `instanceof` compatibility with Node's builtin `Stream`. It wasn't working in the first place and no one complained.

1. If breaking this compatibility is not desired, set `stream: false` for both the `browser` and `react-native` fields in `package.json`. Call `require('stream')` and check that it returns `Stream` and not the empty object before using it.

I can submit a PR for this. 

[stream-false]: https://github.com/nodejs/readable-stream/blob/4dbe6e20051e44025d3cbec3ed9603483d8b27cb/package.json
[browser spec]: https://github.com/defunctzombie/package-browser-field-spec/tree/4f296871cee64e60124841c06c06511885152f19#ignore-a-module
[patch]: https://gist.github.com/parshap/c010022a7f23d6976b6f5ca4905556af
